### PR TITLE
Adds support for custom user id/gid

### PR DESCRIPTION
**Pull request by qbit-**
This adds support for custom user id/gid. Setting custom id is convenient if docker images build with `FROM` are used for code development. This removes troubles with access rights of mounted folders. It also allows to install packages inside the container with non-root id/gid

### CODE REVIEW

**Comment by wholtz** [comment]
Hello @qbit and thank you for your interest in micromamba. 

As noted in the `README.md`, you can use `docker run --user=UID:GID mambaorg/micromamba:latest`  to run with an arbitrary UID/GID combination. 

It is unclear to me what problem you are attempting to solve with this PR. I can write to mounted volumes:
```docker run  -v $(pwd):/mnt --rm --user=1111:1111 mambaorg/micromamba:latest touch /mnt/foobar```
I can install conda packages:
```docker run --rm --user=1111:1111 mambaorg/micromamba:latest micromamba install -y -c conda-forge ca-certificates```
I can't `apt-get install` as UID 1111, but would violate standard unix security policies.

Can you please show me some example commands that you believe should work but currently fail?

**Comment by qbit** [comment]
@qbit-

**Comment by qbit-** [comment]
Hello! Thanks for the discussion, sorry for the longer response time. The problem with setting UID/GID using docker 
keyword is that there is no valid username/group name for the uid/gid you can supply. This results in the 
"I have no name!" string shown in the prompt, and some other unpleasant effects with missing user-related environment variables ($HOME, etc). For example, you can not start `tmux` in this container unless you define all env by hand.
If you know a better way to fix these issues using some clever shell script I would be glad to implement it.
Another use case is hard-code some specific uid/gid if the container is used to run a service

**Comment by qbit-** [comment]
Well, my use case was to build a micromamba container which works well with custom uid/gid and allows to run `ssh` from inside of the container). Additionally I wanted a CUDA-based base image, hence rebuilt `mambaorg/micromamba` from scratch with another base which worked perfectly. Two more 'ARG' do not seem like a major complication to me, but maybe there are reasons why not to do it.  

The same effect with uid/gid can be probably achieved by using `FROM` and modifying the `micromamba` user. I can add a more detailed section in `README.md` instead

**Comment by maresb** [comment]
I have not carefully read this PR, but I simply wanted to point out that a lot of work on this problem has been done in jupyter/docker-stacks, so that may be a useful reference.

Most relevant is this part of the [documentation](https://jupyter-docker-stacks.readthedocs.io/en/latest/using/common.html#user-related-configurations), and then the [startup script](https://github.com/jupyter/docker-stacks/blob/master/base-notebook/start.sh) where most action takes place.

I think things could be done better, so I wouldn't recommend copying. (I haven't read this PR carefully enough, so it's quite possible that you're already doing things better.)

File: Dockerfile [hunk 1]
```diff
@@ -27,11 +27,14 @@ COPY --from=stage1 /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certific
...
 ARG MAMBA_USER=mambauser
+ARG MAMBA_USER_ID=1000
+ARG MAMBA_USER_GID=1000
 ENV MAMBA_USER=$MAMBA_USER
 
 RUN echo "source /usr/local/bin/_activate_current_env.sh" >> ~/.bashrc && \
     echo "source /usr/local/bin/_activate_current_env.sh" >> /etc/skel/.bashrc && \
-    useradd -ms /bin/bash "$MAMBA_USER" && \
+    useradd -u "${MAMBA_USER_ID}" -ms /bin/bash -U "${MAMBA_USER}" && \
+    groupmod -g "${MAMBA_USER_GID}" "${MAMBA_USER}" && \
```
[the remark this answers is not in the document]
Comment (qbit-) [reply Dockerfile]: Ok, I have modified this command and added 2 tests. One for building `micromamba` image with custom id/gid and another one for building a derived image with customized id/gid

**Review by wholtz — APPROVED** [review]

**Pull request merged**